pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1040,3 +1060,25 @@ func (m *ManagerImpl) setPodPendingAdmission(pod *v1.Pod) {
 
 	m.pendingAdmissionPod = pod
 }
+
+func (m *ManagerImpl) checkContainerAlreadyRunning(podUID, cntName string) bool {
+	if m.sourcesReady.AllReady() {
+		// initialization done, let's stop any guessing
+		return false
+	}
+
+	cntID, err := m.containerMap.GetContainerID(podUID, cntName)
+	if err != nil {
+		klog.V(4).InfoS("container not found in the initial map, assumed NOT running", "podUID", podUID, "containerName", cntName, "err", err)
+		return false
+	}
+
+	if !m.containerRunningSet.Has(cntID) {
```

Comment:
So here you check that pod is not starting. Approved.
It a bit odd that Container Map return getContainerID return while RunninSet uses `has` function.
But it is ok that we need only check that the container is running.